src/algebraic_topology/split_simplicial_object.lean

```diff
@@ -0,0 +1,227 @@
+Authors: Joël Riou
+-/
+
+import algebraic_topology.simplicial_object
+import category_theory.limits.shapes.finite_products
+
+/-!
+
+# Split simplicial objects
+
+In this file, we introduce the notion of split simplicial object.
+If `C` is a category that has finite coproducts, a splitting
+`s : splitting X` of a simplical object `X` in `C` consists
+of the datum of a sequence of objects `s.N : ℕ → C` and a
+sequence of morphisms `s.ι n : s.N n → X _[n]` that have
+the property that a certain canonical map identifies `X _[n]`
+with the direct sum of objects `s.N i` indexed by all possible
+epimorphisms `[n] ⟶ [i]` in `simplex_category`. (We do not
+assume that the morphisms `s.ι n` are monomorphisms: in the
+most common categories, this would be a consequence of the
+axioms.)
+
+## References
+* [Stacks: Splitting simplicial objects] https://stacks.math.columbia.edu/tag/017O
+
+-/
+
+noncomputable theory
+
+open category_theory
+open category_theory.category
+open category_theory.limits
+open opposite
+open_locale simplicial
+
+universe u
+
+variables {C : Type*} [category C]
+
+namespace simplicial_object
+
+namespace splitting
+
+/-- The index set which appears in the definition of split simplicial objects. -/
+def index_set (Δ : simplex_categoryᵒᵖ) :=
+Σ (Δ' : simplex_categoryᵒᵖ), { α : Δ.unop ⟶ Δ'.unop // epi α }
+
+namespace index_set
+
+/-- The element in `splitting.index_set Δ` attached to an epimorphism `f : Δ ⟶ Δ'`. -/
+@[simps]
+def mk {Δ Δ' : simplex_category} (f : Δ ⟶ Δ') [epi f] : index_set (op Δ) :=
+⟨op Δ', f, infer_instance⟩
+
+variables {Δ' Δ : simplex_categoryᵒᵖ} (A : index_set Δ)
+
+/-- The epimorphism in `simplex_category` associated to `A : splitting.index_set Δ` -/
+def e := A.2.1
+
+instance : epi A.e := A.2.2
+
+lemma ext' : A = ⟨A.1, ⟨A.e, A.2.2⟩⟩ := by tidy
+
+lemma ext (A₁ A₂ : index_set Δ) (h₁ : A₁.1 = A₂.1)
+  (h₂ : A₁.e ≫ eq_to_hom (by rw h₁) = A₂.e) : A₁ = A₂ :=
+begin
+  rcases A₁ with ⟨Δ₁, ⟨α₁, hα₁⟩⟩,
+  rcases A₂ with ⟨Δ₂, ⟨α₂, hα₂⟩⟩,
+  simp only at h₁,
+  subst h₁,
+  simp only [eq_to_hom_refl, comp_id, index_set.e] at h₂,
+  simp only [h₂],
+end
+
+instance : fintype (index_set Δ) :=
+fintype.of_injective
+  ((λ A, ⟨⟨A.1.unop.len, nat.lt_succ_iff.mpr
+    (simplex_category.len_le_of_epi (infer_instance : epi A.e))⟩, A.e.to_order_hom⟩) :
+    index_set Δ → (sigma (λ (k : fin (Δ.unop.len+1)), (fin (Δ.unop.len+1) → fin (k+1)))))
+begin
+  rintros ⟨Δ₁, α₁⟩ ⟨Δ₂, α₂⟩ h₁,
+  induction Δ₁ using opposite.rec,
+  induction Δ₂ using opposite.rec,
+  simp only at h₁,
+  have h₂ : Δ₁ = Δ₂ := by { ext1, simpa only [subtype.mk_eq_mk] using h₁.1, },
+  subst h₂,
+  refine ext _ _ rfl _,
+  ext : 2,
+  exact eq_of_heq h₁.2,
+end
+
+variable (Δ)
+
+/-- The distinguished element in `splitting.index_set Δ` which corresponds to the
+identity of `Δ`. -/
+def id : index_set Δ := ⟨Δ, ⟨𝟙 _, by apply_instance,⟩⟩
+
+instance : inhabited (index_set Δ) := ⟨id Δ⟩
+
+end index_set
+
+variables (N : ℕ → C) (Δ : simplex_categoryᵒᵖ)
```

Comment:
Is there a benefit in using `nat` to index `N` as opposed to `simplex_category` itself?

Reply:
Thanks for the review! So far, it should not change much to use `nat`, `simplex_category` (or its the opposite category). However, when we relate splittings of simplicial objects and their the filtration by the skeleton, we shall use `boundary_inclusion (n : ℕ) : ∂Δ[n] ⟶ Δ[n]`. Then, eventually, I believe that using `nat` might be more convenient.
(I will do the other changes when I have time to do so.)